modules/core/src/test/java/org/apache/ignite/internal/processors/cache/transactions/TxOnCachesStopTest.java

```diff
@@ -254,10 +345,182 @@ private void runTxOnCacheStop(TransactionConcurrency conc, TransactionIsolation
     /**
      * @throws Exception If failed.
      */
+    @Test
+    public void testOptimisticTransactionsOnCacheDestroy () throws Exception {
+        Assume.assumeFalse(MvccFeatureChecker.forcedMvcc());
+
+        startGridsMultiThreaded(3);
+
+        Ignition.setClientMode(true);
+
+        ArrayList<Ignite> clients = new ArrayList<>();
+        for (int ci = 0; ci < 2; ++ci)
+            clients.add(startGrid("client-" + ci));
+
+        clients.get(0).cluster().active(true);
+
+        List<CacheConfiguration> cachesToBeDestroyed = createCacheConfigurations();
+
+        for (TransactionIsolation iso : TransactionIsolation.values()) {
+            clients.get(0).getOrCreateCaches(cachesToBeDestroyed);
+
+            // Make sure that all caches are started.
+            awaitPartitionMapExchange();
+
+            testConcurrentTransactionsOnCacheDestroy(clients, OPTIMISTIC, iso);
+
+            // Make sure that all caches are stopped.
+            awaitPartitionMapExchange();
+        }
+    }
+
+    /**
+     * Creates a list of cache configurations.
+     *
+     * @return List of cache configurations.
+     */
+    private List<CacheConfiguration> createCacheConfigurations() {
+        String GRP_NAME = "test-destroy-group";
+
+        List<CacheConfiguration> cacheCfgs = new ArrayList<>(50);
```

Comment:
What is 50? You have already CACHE_CNT, could we replace?

Reply:
Nice catch!